svf/include/SVFIR/SVFIR.h

```diff
@@ -709,35 +709,35 @@ class SVFIR : public IRGraph
 
     ///  Add a dummy value/object node according to node ID (llvm value is null)
     //@{
-    inline NodeID addDummyValNode(NodeID i)
+    inline NodeID addDummyValNode(NodeID i, const ICFGNode* node)
     {
-        return addValNode(new DummyValVar(i));
+        return addValNode(new DummyValVar(i, node));
     }
-    inline NodeID addDummyObjNode(NodeID i, const SVFType* type)
+    inline NodeID addDummyObjNode(NodeID i, const SVFType* type, const ICFGNode* node)
     {
         if (symInfo->idToObjTypeInfoMap().find(i) == symInfo->idToObjTypeInfoMap().end())
         {
             ObjTypeInfo* ti = symInfo->createObjTypeInfo(type);
             symInfo->idToObjTypeInfoMap()[i] = ti;
-            return addObjNode(new DummyObjVar(i, ti, type));
+            return addObjNode(new DummyObjVar(i, ti, node, type));
         }
         else
         {
-            return addObjNode(new DummyObjVar(i, symInfo->getObjTypeInfo(i), type));
+            return addObjNode(new DummyObjVar(i, symInfo->getObjTypeInfo(i), node, type));
         }
     }
 
-    inline NodeID addBlackholeObjNode()
+    inline NodeID addBlackholeObjNode(const ICFGNode* node)
```

Comment:
remove the argument here and make the nullptr at the constructor:

new DummyObjVar(getBlackHoleNode(), symInfo->getObjTypeInfo(getBlackHoleNode()), nullptr)